Security/Voter/HandleCommandVoter.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+namespace League\Tactician\Bundle\Security\Voter;
+
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Authorization\AccessDecisionManagerInterface;
+use Symfony\Component\Security\Core\Authorization\Voter\Voter;
+
+/**
+ * Voter for security checks on handling commands.
+ *
+ * @author Ron Rademaker
+ */
+class HandleCommandVoter extends Voter
+{
+    /**
+     * The decision manager.
+     *
+     * @var AccessDecisionManagerInterface
+     */
+    private $decisionManager;
+
+    /**
+     * Default required role.
+     *
+     * @var string
+     */
+    private $defaultRole;
+
+    /**
+     * Command - Require role mapping
+     *
+     * @var array
+     */
+    private $commandRoleMapping;
+
+    /**
+     * Create a new HandleCommandVoter.
+     *
+     * @param AccessDecisionManagerInterface $decisionManager
+     * @param array $commandRoleMapping
+     * @param string $defaultRole
+     */
+    public function __construct(AccessDecisionManagerInterface $decisionManager, array $commandRoleMapping = [], $defaultRole = null)
+    {
+        $this->decisionManager = $decisionManager;
+        $this->commandRoleMapping = $commandRoleMapping;
+        $this->defaultRole = $defaultRole;
+    }
+
+    /**
+     * The voter supports checking handle commands
+     *
+     * @param string $attribute
+     * @param object $subject
+     * @return bool
+     */
+    protected function supports($attribute, $subject)
+    {
+        return $attribute === 'handle' && is_object($subject);
+    }
+
+    /**
+     * Checks if the currently logged on user may handle $subject.
+     *
+     * @param type $attribute
+     * @param type $subject
+     * @param TokenInterface $token
+     *
+     * @return bool
+     */
+    protected function voteOnAttribute($attribute, $subject, TokenInterface $token)
+    {
+        $allowedRoles = $this->getAllowedRoles(get_class($subject));
+
+        if (count($allowedRoles) > 0) {
+            return $this->getDecisionManager()->decide($token, $allowedRoles);
+        } else {
+            // no required role is action allowed
+            return true;
```

Comment:
I'd prefer to see this reject by default. I see security as an optional middleware and if you're going to enable it, you should list every command's allowed roles explicitly.